src/core/qgsvectorlayer.cpp

```diff
@@ -2387,6 +2387,19 @@ long QgsVectorLayer::featureCount() const
          ( mEditBuffer ? mEditBuffer->mAddedFeatures.size() - mEditBuffer->mDeletedFeatureIds.size() : 0 );
 }
 
+long QgsVectorLayer::featureCount( QgsFeatureRequest request )
```

Comment:
Yes it's an helper method to have a simpliest way to count features in Python.
And it's also an helper method for WFS 2.0 implementation in QGIS Server.

If you think it's not useful, I can close this pull-request.